### PR TITLE
Update aliwangwang from 20180413-1510-8.00.44 to 20200723-8.01.03

### PR DESCRIPTION
**Pull request by ElfSundae**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` left no offenses.
- [ ] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask/pulls) for the same update.
- [ ] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Created with [`cask-repair`](https://github.com/Homebrew/homebrew-cask/blob/master/CONTRIBUTING.md#updating-a-cask).

### CODE REVIEW

**Comment by ElfSundae** [comment]
The `appcast` is captured from the "Check Updates" menu item in this app.

File: Casks/aliwangwang.rb [hunk 1]
```diff
@@ -1,9 +1,10 @@
 cask "aliwangwang" do
-  version "20180413-1510-8.00.44"
```
Comment (vitorgalvao) [inline Casks/aliwangwang.rb:2]: I still get this version from the download on the website.

File: Casks/aliwangwang.rb [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
-  sha256 "06de2d82ffb839350ee54bd16227f910b7709563f86ceb67d32ff0e050fb6515"
+  version "20200723-8.01.03"
+  sha256 "c7ed5cc9ace67b6ba65ee84dae24cb467e36118359b7538762036071ead0ba05"
 
-  url "https://alimarket.taobao.com/markets/qnww/aliww-download?wh_from=macos"
-  appcast "https://macupdater.net/cgi-bin/check_urls/check_url_filename.cgi?url=https://alimarket.taobao.com/markets/qnww/aliww-download?wh_from=macos"
+  # gw.alipayobjects.com was verified as official when first introduced to the cask
+  url "https://gw.alipayobjects.com/os/bmw-prod/9b3facdd-e15f-4013-a883-b48ef641d700.dmg"
+  appcast "https://jdy.tmall.com/version/check?version=8.00.44M&nick=cask&platform=macww"
```
Comment (vitorgalvao) [inline Casks/aliwangwang.rb:7]: It has a specific version, which makes it way less useful. Th trick in these cases is to try something different, like `0` or `0.0.0`. In this case, `8` works. `version` needs to be changed so we can somehow use that.

File: Casks/aliwangwang.rb [hunk 3]
```diff
@@ -1,9 +1,10 @@
 cask "aliwangwang" do
-  version "20180413-1510-8.00.44"
-  sha256 "06de2d82ffb839350ee54bd16227f910b7709563f86ceb67d32ff0e050fb6515"
+  version "20200723-8.01.03"
+  sha256 "c7ed5cc9ace67b6ba65ee84dae24cb467e36118359b7538762036071ead0ba05"
 
-  url "https://alimarket.taobao.com/markets/qnww/aliww-download?wh_from=macos"
-  appcast "https://macupdater.net/cgi-bin/check_urls/check_url_filename.cgi?url=https://alimarket.taobao.com/markets/qnww/aliww-download?wh_from=macos"
+  # gw.alipayobjects.com was verified as official when first introduced to the cask
+  url "https://gw.alipayobjects.com/os/bmw-prod/9b3facdd-e15f-4013-a883-b48ef641d700.dmg"
```
Comment (vitorgalvao) [inline Casks/aliwangwang.rb:6]: How are you getting at this particular URL?

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

File: Casks/aliwangwang.rb (hunk 1 above)
Comment (ElfSundae) [reply Casks/aliwangwang.rb:2]: Yes, the website hasn't been updated for a long time. The new version is prompted when launching the app.
image

**Comment by ElfSundae** [comment]
@vitorgalvao Thanks for your review.

I captured this app's network traffic using Charles.

image
image
image

The "Check for Updates" (检查更新) menu item requests `http://jdy.tmall.com/version/check?version=8.00.44M&nick=baidukissgoogle&platform=macww&`, and gets the following response:
```json
{
  "version_check_get_response": {
    "version": "8.01.03M",
    "title": null,
    "message": "解决文件传输问题",
    "downloadUrl": "https://gw.alipayobjects.com/os/bmw-prod/9b3facdd-e15f-4013-a883-b48ef641d700.dmg",
    "jailBrokeDownloadUrl": null,
    "aimUrl": null,
    "guideType": 1,
    "forceUpdate": null,
    "feature": null
  }
}
```

By this way, I get the new `appcast` and the new download `url`.

**Comment by ElfSundae** [comment]
And the new version `20200723-8.01.03`, could be got by download the `url`.

```bash
curl -I https://gw.alipayobjects.com/os/bmw-prod/9b3facdd-e15f-4013-a883-b48ef641d700.dmg
HTTP/1.1 200 Connection established

HTTP/2 200 
server: Tengine
content-type: application/octet-stream
content-length: 73179206
date: Sun, 11 Oct 2020 11:44:33 GMT
x-oss-request-id: 5F82F021B1445536342F727B
x-oss-cdn-auth: success
accept-ranges: bytes
etag: "A1ECD941592F5D4BB4290C0FE454CF81"
last-modified: Thu, 23 Jul 2020 07:44:50 GMT
x-oss-object-type: Normal
x-oss-hash-crc64ecma: 7714528076251814280
x-oss-storage-class: Standard
x-oss-meta-file-type: dmg
x-oss-meta-filename: 9b3facdd-e15f-4013-a883-b48ef641d700.dmg
content-disposition: attachment; filename="Aliwangwang-20200723-8.01.03.dmg"; filename*=UTF-8''Aliwangwang-20200723-8.01.03.dmg
```

**Comment by vitorgalvao** [comment]
Both the `appcast` and `url` are from different hosts than the `homepage`. Coupled with the facts the app and website are in Chinese (which HBC maintainers don’t read), the website doesn’t list this version, and we need to resort to network sniffing to even get a new URL, this cask is very difficult to audit.

For now, we’ll need to keep this at the current version. Upstream needs to get it together and update the website, or provide links that aren’t from wildly different hosts, or we’ll need to remove the cask (which is likely the safest course of action, seeing as it’s outdated by over two years).

I have no reason to distrust you, but our users may not feel the same. We have precedent of not keeping casks that can’t be easily audited (i.e. without having to download the software).

**Pull request closed without merging**